packages/mongodb-chatbot-ui/src/Message.tsx

```diff
@@ -260,3 +332,110 @@ export const MessagePrompts = ({
     </CSSTransition>
   );
 };
+
+export type MessageRatingWithFeedbackCommentProps = MessageRatingProps & {
+  submit: (commentText: string) => void;
+  abandon: () => void;
+  status: "none" | "submitted" | "abandoned";
+  errorMessage?: string;
+};
+
+export function MessageRatingWithFeedbackComment(
+  props: MessageRatingWithFeedbackCommentProps
+) {
+  const {
+    value,
+    submit,
+    abandon,
+    status,
+    errorMessage,
+    ...messageRatingProps
+  } = props;
+
+  const hasRating = value !== undefined;
+
+  // TODO: Use this to animate the transition after https://jira.mongodb.org/browse/LG-3965 is merged.
+  // const ratingCommentInputVisible = hasRating && status !== "submitted" && status !== "abandoned";
+
+  const ratingCommentRef = useRef(null);
+
+  const isSubmitted = status === "submitted";
+  console.log("isSubmitted", isSubmitted);
+
+  return (
+    <div
+      // TODO: This css is a hacky fix until https://jira.mongodb.org/browse/LG-3965 is merged
+      // Once merged we can apply this margin directly to InlineMesssageFeedback
+      className={css`
+        & > div + div {
+          margin-top: 0.5rem;
+        }
+      `}
+    >
+      <MessageRating value={value} {...messageRatingProps} />
+      {hasRating && status !== "abandoned" ? (
+        <>
+          <InlineMessageFeedback
```

Comment:
can you add a max input length FE validation to the text area? presumably same as what's on the server. 

if server is configurable, then i guess that length should be here too.

Reply:
It's a little hacky since this isn't baked into the LG component but it works!